charmcraft/models/actions.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Charmcraft Juju Actions pydantic model."""
+
+import re
+import keyword
+
+import pydantic
+
+from typing import Optional, Dict
+
+from charmcraft.models.basic import ModelConfigDefaults
+
+
+class JujuActions(ModelConfigDefaults):
+    """Juju actions for charms.
+
+    See also: https://juju.is/docs/sdk/actions
+    """
+
+    _action_name_regex = re.compile(r"^[a-zA-Z_][a-zA-Z0-9-_]*$")
+    actions: Optional[Dict[str, Dict]]
+    legacy: bool = False
+
+    @pydantic.validator("actions", pre=True)
```

Comment:
Why does this need to be a pre validator? (Please include the reasoning in the docstring or a comment so it doesn't get lost to time)

Reply:
I don't think it need `pre` in this case.